src/app/buyeroffer/buyeroffer.ts

```diff
@@ -1,6 +1,10 @@
+import {Advertisement} from "../advertisement/advertisement";
+
 export class BuyerOffer {
   uri: string = '';
-  buyer_id: number;
+  /*advOffers: string;*/
+  advertisement: string;
```

Comment:
Why are you storing advertisement as a string? It should be `advertisement: Advertisement` instead.

The one you are importing `import { Advertisement } from "../advertisement/advertisement";` is the class that should store everything you need:
- `id` (automatically generated by entering the `uri`)
- `title`
- seller (as `owner`)

By using `Advertisement` you will have easy access to all advertisement services, and your `BuyerOffer` class will be easier to reuse.